// The following messages are used to start OSQuery flows in GRR.

syntax = "proto2";

import "grr_response_proto/semantic.proto";

message OSQueryScheduledQueryArgs {
  required string query_id = 1 [(sem_type) = {
      description: "Query ID"
    }];
  required string query = 2 [(sem_type) = {
      description: "OSQuery SQL statement; passed as a string."
    }];
  required uint32 interval = 3 [(sem_type) = {
      description: "interval in seconds"
    }];
}

message OSQueryScheduledQueryResult {
  optional string status = 1;
}


message OSQueryRunQueryArgs {
  optional string query = 1 [(sem_type) = {
      description: "OSQuery SQL statement; passed as a string."
    }];
}

message OSQueryRemoveScheduledQueryArgs {
  optional string query_id = 1 [(sem_type) = {
      description: "Query ID"
    }];
}

message OSQueryRemoveScheduledQueryResult {
  optional string status = 1;
}

message OSQueryListScheduledQueriesResult {
  optional string query_id = 1;
  optional string query = 2;
  optional uint32 interval = 3;
}

message OSQueryPullScheduledQueryArgs {
  optional string query_id = 1 [(sem_type) = {
      description: "Query ID"
    }];
}

message OSQueryPullScheduledQueryResult {
  optional string name = 1;
  optional string calendarTime = 2;
  optional string counter = 3;
  optional string epoch = 4;
  optional string unixTime = 5;
  optional string action = 6;
  optional OSQueryRunQueryResult row = 7;
}

message OSQueryRunQueryResult {
  optional string error_msg = 1;
  repeated string access_time = 2;
  repeated string account_id = 3;
  repeated string action = 4;
  repeated string active = 5;
  repeated string active_disks = 6;
  repeated string address = 7;
  repeated string algorithm = 8;
  repeated string alias = 9;
  repeated string aliases = 10;
  repeated string ami_id = 11;
  repeated string api_version = 12;
  repeated string arch = 13;
  repeated string architecture = 14;
  repeated string args = 15;
  repeated string arguments = 16;
  repeated string atime = 17;
  repeated string attached = 18;
  repeated string auid = 19;
  repeated string author = 20;
  repeated string autoupdate = 21;
  repeated string availability_zone = 22;
  repeated string average = 23;
  repeated string average_memory = 24;
  repeated string base_uri = 25;
  repeated string binary_queue = 26;
  repeated string bitmap_chunk_size = 27;
  repeated string bitmap_external_file = 28;
  repeated string bitmap_on_mem = 29;
  repeated string blacklisted = 30;
  repeated string blocks = 31;
  repeated string blocks_available = 32;
  repeated string blocks_free = 33;
  repeated string block_size = 34;
  repeated string blocks_size = 35;
  repeated string bridge_nf_ip6tables = 36;
  repeated string bridge_nf_iptables = 37;
  repeated string broadcast = 38;
  repeated string btime = 39;
  repeated string buffers = 40;
  repeated string build = 41;
  repeated string build_distro = 42;
  repeated string build_platform = 43;
  repeated string build_time = 44;
  repeated string bytes = 45;
  repeated string cached = 46;
  repeated string carve = 47;
  repeated string carve_guid = 48;
  repeated string category = 49;
  repeated string cgroup_driver = 50;
  repeated string chain = 51;
  repeated string change_time = 52;
  repeated string chassis_bridge_capability_available = 53;
  repeated string chassis_bridge_capability_enabled = 54;
  repeated string chassis_docsis_capability_available = 55;
  repeated string chassis_docsis_capability_enabled = 56;
  repeated string chassis_id = 57;
  repeated string chassis_id_type = 58;
  repeated string chassis_mgmt_ips = 59;
  repeated string chassis_other_capability_available = 60;
  repeated string chassis_other_capability_enabled = 61;
  repeated string chassis_repeater_capability_available = 62;
  repeated string chassis_repeater_capability_enabled = 63;
  repeated string chassis_router_capability_available = 64;
  repeated string chassis_router_capability_enabled = 65;
  repeated string chassis_station_capability_available = 66;
  repeated string chassis_station_capability_enabled = 67;
  repeated string chassis_sys_description = 68;
  repeated string chassis_sysname = 69;
  repeated string chassis_tel_capability_available = 70;
  repeated string chassis_tel_capability_enabled = 71;
  repeated string chassis_wlan_capability_available = 72;
  repeated string chassis_wlan_capability_enabled = 73;
  repeated string check_array_finish = 74;
  repeated string check_array_progress = 75;
  repeated string check_array_speed = 76;
  repeated string chunk_size = 77;
  repeated string class = 78;
  repeated string cmdline = 79;
  repeated string cmdline_size = 80;
  repeated string codename = 81;
  repeated string collect_cross_processes = 82;
  repeated string collect_data_file_writes = 83;
  repeated string collect_emet_events = 84;
  repeated string collect_file_mods = 85;
  repeated string collect_module_info = 86;
  repeated string collect_module_loads = 87;
  repeated string collect_net_conns = 88;
  repeated string collect_processes = 89;
  repeated string collect_process_user_context = 90;
  repeated string collect_reg_mods = 91;
  repeated string collect_sensor_operations = 92;
  repeated string collect_store_files = 93;
  repeated string collisions = 94;
  repeated string command = 95;
  repeated string comment = 96;
  repeated string common_name = 97;
  repeated string component = 98;
  repeated string computer_name = 99;
  repeated string config_hash = 100;
  repeated string config_name = 101;
  repeated string config_valid = 102;
  repeated string config_value = 103;
  repeated string connection_id = 104;
  repeated string connection_status = 105;
  repeated string containers = 106;
  repeated string containers_paused = 107;
  repeated string containers_running = 108;
  repeated string containers_stopped = 109;
  repeated string core = 110;
  repeated string count = 111;
  repeated string cpu = 112;
  repeated string cpu_brand = 113;
  repeated string cpu_cfs_period = 114;
  repeated string cpu_cfs_quota = 115;
  repeated string cpu_kernelmode_usage = 116;
  repeated string cpu_logical_cores = 117;
  repeated string cpu_physical_cores = 118;
  repeated string cpus = 119;
  repeated string cpu_set = 120;
  repeated string cpu_shares = 121;
  repeated string cpu_subtype = 122;
  repeated string cpu_total_usage = 123;
  repeated string cpu_type = 124;
  repeated string cpu_usermode_usage = 125;
  repeated string created = 126;
  repeated string create_time = 127;
  repeated string creator = 128;
  repeated string creator_pid = 129;
  repeated string creator_uid = 130;
  repeated string ctime = 131;
  repeated string current_value = 132;
  repeated string cwd = 133;
  repeated string data = 134;
  repeated string date = 135;
  repeated string datetime = 136;
  repeated string date_time = 137;
  repeated string day = 138;
  repeated string day_of_month = 139;
  repeated string day_of_week = 140;
  repeated string days = 141;
  repeated string default_value = 142;
  repeated string description = 143;
  repeated string destination = 144;
  repeated string device = 145;
  repeated string device_alias = 146;
  repeated string device_name = 147;
  repeated string dhcp_enabled = 148;
  repeated string dhcp_lease_expires = 149;
  repeated string dhcp_lease_obtained = 150;
  repeated string dhcp_server = 151;
  repeated string directory = 152;
  repeated string disabled = 153;
  repeated string discovery_cache_hits = 154;
  repeated string discovery_executions = 155;
  repeated string disk_read = 156;
  repeated string disk_write = 157;
  repeated string dns_domain = 158;
  repeated string dns_domain_suffix_search_order = 159;
  repeated string dns_host_name = 160;
  repeated string dns_server_search_order = 161;
  repeated string drive_name = 162;
  repeated string driver = 163;
  repeated string dst_ip = 164;
  repeated string dst_mask = 165;
  repeated string dst_port = 166;
  repeated string dtime = 167;
  repeated string eapi = 168;
  repeated string egid = 169;
  repeated string eid = 170;
  repeated string enabled = 171;
  repeated string enable_ipv6 = 172;
  repeated string encrypted = 173;
  repeated string end = 174;
  repeated string endpoint_id = 175;
  repeated string env = 176;
  repeated string env_count = 177;
  repeated string environment = 178;
  repeated string environment_count = 179;
  repeated string environment_size = 180;
  repeated string env_size = 181;
  repeated string euid = 182;
  repeated string event = 183;
  repeated string event_queue = 184;
  repeated string events = 185;
  repeated string executions = 186;
  repeated string expire = 187;
  repeated string extensions = 188;
  repeated string extra = 189;
  repeated string facility = 190;
  repeated string failed_disks = 191;
  repeated string family = 192;
  repeated string fd = 193;
  repeated string feature = 194;
  repeated string feature_control = 195;
  repeated string filename = 196;
  repeated string filter_name = 197;
  repeated string flag = 198;
  repeated string flags = 199;
  repeated string friendly_name = 200;
  repeated string gateway = 201;
  repeated string gid = 202;
  repeated string gid_signed = 203;
  repeated string git_commit = 204;
  repeated string go_version = 205;
  repeated string groupname = 206;
  repeated string group_sid = 207;
  repeated string guest = 208;
  repeated string guest_nice = 209;
  repeated string handle = 210;
  repeated string hard_links = 211;
  repeated string hardware_model = 212;
  repeated string hardware_serial = 213;
  repeated string hardware_vendor = 214;
  repeated string hardware_version = 215;
  repeated string hash_alg = 216;
  repeated string hashed = 217;
  repeated string header = 218;
  repeated string header_size = 219;
  repeated string history_file = 220;
  repeated string host = 221;
  repeated string host_ip = 222;
  repeated string hostname = 223;
  repeated string hostnames = 224;
  repeated string host_port = 225;
  repeated string hour = 226;
  repeated string hours = 227;
  repeated string http_proxy = 228;
  repeated string https_proxy = 229;
  repeated string iam_arn = 230;
  repeated string ibytes = 231;
  repeated string id = 232;
  repeated string identifier = 233;
  repeated string idle = 234;
  repeated string idrops = 235;
  repeated string ierrors = 236;
  repeated string image = 237;
  repeated string image_id = 238;
  repeated string images = 239;
  repeated string inactive = 240;
  repeated string iniface = 241;
  repeated string iniface_mask = 242;
  repeated string inode = 243;
  repeated string inodes = 244;
  repeated string inodes_free = 245;
  repeated string input_eax = 246;
  repeated string instance_id = 247;
  repeated string instance_type = 248;
  repeated string interface = 249;
  repeated string internal = 250;
  repeated string interval = 251;
  repeated string iowait = 252;
  repeated string ipackets = 253;
  repeated string ip_address = 254;
  repeated string ip_prefix_len = 255;
  repeated string ipv4_forwarding = 256;
  repeated string ipv6_address = 257;
  repeated string ipv6_gateway = 258;
  repeated string ipv6_prefix_len = 259;
  repeated string irq = 260;
  repeated string iso_8601 = 261;
  repeated string issuer_common_name = 262;
  repeated string issuer_organization = 263;
  repeated string issuer_organization_unit = 264;
  repeated string kernel_memory = 265;
  repeated string kernel_version = 266;
  repeated string key = 267;
  repeated string key_file = 268;
  repeated string keyword = 269;
  repeated string label = 270;
  repeated string last_change = 271;
  repeated string last_executed = 272;
  repeated string license = 273;
  repeated string local_address = 274;
  repeated string locale = 275;
  repeated string local_hostname = 276;
  repeated string local_ipv4 = 277;
  repeated string local_port = 278;
  repeated string localtime = 279;
  repeated string local_time = 280;
  repeated string local_timezone = 281;
  repeated string location = 282;
  repeated string locked = 283;
  repeated string log_file_disk_quota_mb = 284;
  repeated string log_file_disk_quota_percentage = 285;
  repeated string logging_driver = 286;
  repeated string mac = 287;
  repeated string mac_address = 288;
  repeated string maintainer = 289;
  repeated string major = 290;
  repeated string manufacturer = 291;
  repeated string mask = 292;
  repeated string match = 293;
  repeated string matches = 294;
  repeated string max = 295;
  repeated string md5 = 296;
  repeated string md_device_name = 297;
  repeated string med_capability_capabilities = 298;
  repeated string med_capability_inventory = 299;
  repeated string med_capability_location = 300;
  repeated string med_capability_mdi_pd = 301;
  repeated string med_capability_mdi_pse = 302;
  repeated string med_capability_policy = 303;
  repeated string med_device_type = 304;
  repeated string med_policies = 305;
  repeated string mem = 306;
  repeated string memory = 307;
  repeated string memory_free = 308;
  repeated string memory_limit = 309;
  repeated string memory_max_usage = 310;
  repeated string memory_total = 311;
  repeated string memory_usage = 312;
  repeated string message = 313;
  repeated string method = 314;
  repeated string metric = 315;
  repeated string metric_name = 316;
  repeated string metric_value = 317;
  repeated string mime_encoding = 318;
  repeated string mime_type = 319;
  repeated string min = 320;
  repeated string min_api_version = 321;
  repeated string minor = 322;
  repeated string minute = 323;
  repeated string minutes = 324;
  repeated string mode = 325;
  repeated string model = 326;
  repeated string model_id = 327;
  repeated string modify_time = 328;
  repeated string month = 329;
  repeated string mount_point = 330;
  repeated string mtime = 331;
  repeated string mtu = 332;
  repeated string name = 333;
  repeated string native = 334;
  repeated string netmask = 335;
  repeated string network_id = 336;
  repeated string network_rx_bytes = 337;
  repeated string network_tx_bytes = 338;
  repeated string nice = 339;
  repeated string node = 340;
  repeated string no_proxy = 341;
  repeated string nr_raid_disks = 342;
  repeated string number = 343;
  repeated string num_procs = 344;
  repeated string obytes = 345;
  repeated string odrops = 346;
  repeated string oerrors = 347;
  repeated string offset = 348;
  repeated string oid = 349;
  repeated string on_disk = 350;
  repeated string online_cpus = 351;
  repeated string oom_kill_disable = 352;
  repeated string opackets = 353;
  repeated string options = 354;
  repeated string organization = 355;
  repeated string organization_unit = 356;
  repeated string os = 357;
  repeated string os_type = 358;
  repeated string other = 359;
  repeated string outiface = 360;
  repeated string outiface_mask = 361;
  repeated string output_bit = 362;
  repeated string output_register = 363;
  repeated string output_size = 364;
  repeated string overflows = 365;
  repeated string owner_gid = 366;
  repeated string owner_uid = 367;
  repeated string owner_uuid = 368;
  repeated string package = 369;
  repeated string package_cache_file = 370;
  repeated string packets = 371;
  repeated string parent = 372;
  repeated string partition = 373;
  repeated string password_status = 374;
  repeated string patch = 375;
  repeated string path = 376;
  repeated string pci_class = 377;
  repeated string pci_slot = 378;
  repeated string perf_ctl = 379;
  repeated string perf_status = 380;
  repeated string period = 381;
  repeated string permanent = 382;
  repeated string permissions = 383;
  repeated string persistent = 384;
  repeated string pgroup = 385;
  repeated string phys_footprint = 386;
  repeated string physical_adapter = 387;
  repeated string physical_memory = 388;
  repeated string pid = 389;
  repeated string pids = 390;
  repeated string platform = 391;
  repeated string platform_info = 392;
  repeated string platform_like = 393;
  repeated string point_to_point = 394;
  repeated string policy = 395;
  repeated string port = 396;
  repeated string port_aggregation_id = 397;
  repeated string port_autoneg_1000baset_fd_enabled = 398;
  repeated string port_autoneg_1000baset_hd_enabled = 399;
  repeated string port_autoneg_1000basex_fd_enabled = 400;
  repeated string port_autoneg_1000basex_hd_enabled = 401;
  repeated string port_autoneg_100baset2_fd_enabled = 402;
  repeated string port_autoneg_100baset2_hd_enabled = 403;
  repeated string port_autoneg_100baset4_fd_enabled = 404;
  repeated string port_autoneg_100baset4_hd_enabled = 405;
  repeated string port_autoneg_100basetx_fd_enabled = 406;
  repeated string port_autoneg_100basetx_hd_enabled = 407;
  repeated string port_autoneg_10baset_fd_enabled = 408;
  repeated string port_autoneg_10baset_hd_enabled = 409;
  repeated string port_autoneg_enabled = 410;
  repeated string port_autoneg_supported = 411;
  repeated string port_description = 412;
  repeated string port_id = 413;
  repeated string port_id_type = 414;
  repeated string port_mau_type = 415;
  repeated string port_mfs = 416;
  repeated string port_ttl = 417;
  repeated string power_8023at_enabled = 418;
  repeated string power_8023at_power_allocated = 419;
  repeated string power_8023at_power_priority = 420;
  repeated string power_8023at_power_requested = 421;
  repeated string power_8023at_power_source = 422;
  repeated string power_8023at_power_type = 423;
  repeated string power_class = 424;
  repeated string power_device_type = 425;
  repeated string power_mdi_enabled = 426;
  repeated string power_mdi_supported = 427;
  repeated string power_paircontrol_enabled = 428;
  repeated string power_pairs = 429;
  repeated string ppvids_enabled = 430;
  repeated string ppvids_supported = 431;
  repeated string pre_cpu_kernelmode_usage = 432;
  repeated string pre_cpu_total_usage = 433;
  repeated string pre_cpu_usermode_usage = 434;
  repeated string pre_online_cpus = 435;
  repeated string preread = 436;
  repeated string pre_system_cpu_usage = 437;
  repeated string processor_number = 438;
  repeated string propagation = 439;
  repeated string protection_disabled = 440;
  repeated string protocol = 441;
  repeated string pseudo = 442;
  repeated string publisher = 443;
  repeated string pvid = 444;
  repeated string query = 445;
  repeated string raid_disks = 446;
  repeated string raid_level = 447;
  repeated string rapl_energy_status = 448;
  repeated string rapl_power_limit = 449;
  repeated string rapl_power_units = 450;
  repeated string read = 451;
  repeated string recovery_finish = 452;
  repeated string recovery_progress = 453;
  repeated string recovery_speed = 454;
  repeated string refreshes = 455;
  repeated string region = 456;
  repeated string registry = 457;
  repeated string release = 458;
  repeated string remote_address = 459;
  repeated string remote_port = 460;
  repeated string removable = 461;
  repeated string repository = 462;
  repeated string reservation_id = 463;
  repeated string reshape_finish = 464;
  repeated string reshape_progress = 465;
  repeated string reshape_speed = 466;
  repeated string resident_size = 467;
  repeated string response_code = 468;
  repeated string result = 469;
  repeated string resync_finish = 470;
  repeated string resync_progress = 471;
  repeated string resync_speed = 472;
  repeated string revision = 473;
  repeated string rid = 474;
  repeated string root = 475;
  repeated string root_dir = 476;
  repeated string round_trip_time = 477;
  repeated string rule_details = 478;
  repeated string rw = 479;
  repeated string sdk_version = 480;
  repeated string seconds = 481;
  repeated string security_groups = 482;
  repeated string sensor_backend_server = 483;
  repeated string sensor_id = 484;
  repeated string sensor_ip_addr = 485;
  repeated string serial = 486;
  repeated string serial_number = 487;
  repeated string server_version = 488;
  repeated string severity = 489;
  repeated string sgid = 490;
  repeated string sha1 = 491;
  repeated string sha1_fingerprint = 492;
  repeated string sha256 = 493;
  repeated string sha256_fingerprint = 494;
  repeated string shard = 495;
  repeated string shell = 496;
  repeated string shell_only = 497;
  repeated string shmid = 498;
  repeated string sigfile = 499;
  repeated string sig_group = 500;
  repeated string site = 501;
  repeated string size = 502;
  repeated string size_bytes = 503;
  repeated string slot = 504;
  repeated string socket = 505;
  repeated string softirq = 506;
  repeated string source = 507;
  repeated string source_url = 508;
  repeated string spare_disks = 509;
  repeated string speed = 510;
  repeated string src_ip = 511;
  repeated string src_mask = 512;
  repeated string src_port = 513;
  repeated string ssh_public_key = 514;
  repeated string start = 515;
  repeated string start_time = 516;
  repeated string state = 517;
  repeated string status = 518;
  repeated string steal = 519;
  repeated string storage_driver = 520;
  repeated string strings = 521;
  repeated string subclass = 522;
  repeated string subnet = 523;
  repeated string subscriptions = 524;
  repeated string subsystem = 525;
  repeated string success = 526;
  repeated string suid = 527;
  repeated string summary = 528;
  repeated string superblock_state = 529;
  repeated string superblock_update_time = 530;
  repeated string superblock_version = 531;
  repeated string swap_cached = 532;
  repeated string swap_free = 533;
  repeated string swap_limit = 534;
  repeated string swap_total = 535;
  repeated string sycall_addr_modified = 536;
  repeated string symlink = 537;
  repeated string system = 538;
  repeated string system_cpu_usage = 539;
  repeated string system_time = 540;
  repeated string tag = 541;
  repeated string tags = 542;
  repeated string target = 543;
  repeated string target_name = 544;
  repeated string target_path = 545;
  repeated string terminal = 546;
  repeated string text_segment_hash = 547;
  repeated string threads = 548;
  repeated string time = 549;
  repeated string timestamp = 550;
  repeated string timestamp_ms = 551;
  repeated string timezone = 552;
  repeated string total_seconds = 553;
  repeated string total_size = 554;
  repeated string transaction_id = 555;
  repeated string tty = 556;
  repeated string turbo_disabled = 557;
  repeated string turbo_ratio_limit = 558;
  repeated string type = 559;
  repeated string uid = 560;
  repeated string uid_signed = 561;
  repeated string unix_time = 562;
  repeated string unmask = 563;
  repeated string unused_devices = 564;
  repeated string update_url = 565;
  repeated string uptime = 566;
  repeated string url = 567;
  repeated string usb_address = 568;
  repeated string usb_port = 569;
  repeated string use = 570;
  repeated string used_by = 571;
  repeated string user = 572;
  repeated string user_agent = 573;
  repeated string username = 574;
  repeated string user_time = 575;
  repeated string user_uuid = 576;
  repeated string uuid = 577;
  repeated string valid_from = 578;
  repeated string valid_to = 579;
  repeated string value = 580;
  repeated string vendor = 581;
  repeated string vendor_id = 582;
  repeated string version = 583;
  repeated string visible = 584;
  repeated string vlans = 585;
  repeated string volume_size = 586;
  repeated string wall_time = 587;
  repeated string warning = 588;
  repeated string watcher = 589;
  repeated string weekday = 590;
  repeated string wired_size = 591;
  repeated string working_disks = 592;
  repeated string world = 593;
  repeated string year = 594;
}
